teensy/teensy_main/teensy_main.ino

```diff
@@ -428,6 +391,14 @@ void pull_data(){
     set_sensor_status_flags(gps, imu, tpa);
 
     hdr = fill_in_hdr_array(hdr);
+    
+    // Will only sample once the system is initialized.
+    if (SEND_DATA)
+    {
+      //______________Send Data______________//
+      serializeJson(doc, piSerial);
```

Comment:
This is, in effect, the "main" thing the Teensy does from the Raspi's perspective. The main functionality really shouldn't be a sub-part of another big function with a condition checking whether or not to do it.